### ISSUE TITLE
[Feature Request] Allow Smart Playlists to be created for Podcasts

### ISSUE THREAD

**Issue by troycarpenter** [ampache/ampache#3312]
Every morning, I create a play-queue with the latest Podcasts to which I listen to.  It would be nice if I could create a Smart Playlist that would auto create based on criteria I select.

For instance:
Rule1: Podcast Title
Rule2: Newest Episode
Rule3: Hasn't been listened to (by me...I don't know if the database keeps track of per-user listens).

With the proper rules, I would expect a playlist that contains the latest podcast episodes which I can then load and go every morning.

**Comment by lachlan-00** [comment]
Smartlists are just searches hardcoded to song types. it would take a bit to return the type of object correctly and require a bit of expansion as the assumption in most things is that a smartlist is always a song

There is currently not a type for searches using podcast_episodes so that would need to be added. but expanding search for podcasts is easy enough to do.
https://ampache.org/api/api-advanced-search/

**Comment by lachlan-00** [comment]
oh and another thing we need to do is make the podcast_episode table the same as the song table. they're close to songs but not quite so they can't be mashed together without a DB change

**Comment by troycarpenter** [comment]
> oh and another thing we need to do is make the podcast_episode table the same as the song table. they're close to songs but not quite so they can't be mashed together without a DB change

Are you suggesting that a possible solution going forward is that songs and podcasts are all kept in the same table?

**Comment by lachlan-00** [comment]
no, but there are song columns that are needed in the podcast_episode table.

so when displaying a 'media' / playable item they have the same data

**Comment by lachlan-00** [comment]
Haven't commit this yet but this is how the process is going which might make it easier to follow.

Here I've added the podcast_episode into the search class. Then added it into the UI templates
![image](https://user-images.githubusercontent.com/1305249/183320230-27fdb775-6c34-4c74-8d84-ed7875a4131c.png)

Searching creates a tmp_browse row which stores the results.
![image](https://user-images.githubusercontent.com/1305249/183320753-a14b3aad-5ba6-4b41-9339-bd262dee4bb7.png)

this is then meant to be displayed into the search rows. (i haven't fixed the object display yet so it's sending null rows to the results page right now)

To make podcast_episode and song interchangeable into smartlists. they need to match data with the song searches
![image](https://user-images.githubusercontent.com/1305249/183320879-8904c071-f33d-4785-b980-51bd1ec8d3a7.png)

Adding as a search type is pretty easy though but because smartlists are actually just saved song searches it requires more work.

**Comment by lachlan-00** [comment]
collections in ampache 8 let you put in anything and are in the same areas as playlists
